File: app/src/main/java/org/odddev/fantlab/core/network/NetworkChecker.java
package org.odddev.fantlab.core.network;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import org.odddev.fantlab.core.di.Injector;

import javax.inject.Inject;

/**
 * @author kenrube
 * @since 23.08.16
 */

public class NetworkChecker implements INetworkChecker {

    @Inject
    Context mContext;

    public NetworkChecker() {
        Injector.getAppComponent().inject(this);
    }

    @Override
    public boolean isConnected() {
        ConnectivityManager cm = (ConnectivityManager) mContext.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo ni = cm.getActiveNetworkInfo();

        return ni != null && ni.isConnected();
    }
}

File: app/src/main/kotlin/org/odddev/fantlab/core/network/NetworkChecker.kt
package org.odddev.fantlab.core.network

import android.content.Context
import android.net.ConnectivityManager

import org.odddev.fantlab.core.di.Injector

import javax.inject.Inject

class NetworkChecker : INetworkChecker {

	@Inject
	lateinit var context: Context

	init {
		Injector.getAppComponent().inject(this)
	}

	override fun isConnected(): Boolean {
		val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
		val ni = cm.activeNetworkInfo

		return ni?.isConnected ?: false
	}
}
